content/investigating-lang-ext-semantics/index.md

```diff
@@ -0,0 +1,211 @@
+
+The [Haskell Foundation Stability Working Group](https://blog.haskell.org/stability-working-group/) 
+is interested in understanding
+why breakage occurs.
+This is an extension of our initial [investigation](https://jappie.me/analyzing-haskell-stability.html).
+We've recently done further analysis on [head.hackage](https://ghc.gitlab.haskell.org/head.hackage/),
+and learned surprisingly enough that the root cause
+of a lot of breakage isn't Template Haskell,
+but seems to be from language extension semantics[^meaning].
+We're doing this investigation to better understand where the Haskell
+Foundation stability working group should focus it's efforts.
+About a year ago I started on analyzing the causes
+for all `head.hackage` patches, but I got bogged down
+by the sheer amount of work.
+Trevis suggested focusing on the
+most important question,
+why do language extension semantics cause so much breakage?
+So instead of being bogged down by analyzing all
+the patches, I just looked at the 12 patches from language extension
+semantics.
+
+[^meaning]: The precise meaning of features enabled by language extensions. I guess parser changes also count.
+
+This gave us the following table:
+
+| Name                                | Cause                                        | Had warnings? |
+|-------------------------------------+----------------------------------------------+---------------|
+| Cabal-2.4.1.0.patch                 | simplified subsumption                       | no            |
+| Cabal-3.0.2.0.patch                 | simplified subsumption                       | no            |
+| Cabal-3.2.1.0.patch                 | simplified subsumption                       | no            |
+| data-r-tree-0.6.0.patch             | parser change (see 1)                        | no            |
+| drinkery-0.4.patch                  | simplified subsumption                       | no            |
+| ghc-lib-parser-9.8.1.20231121.patch | rename forall identifiers (2)                | yes           |
+| hgeometry-ipe-0.13.patch            | Instances moved due to splice enforcement    | no            |
+| singletons-3.0.2.patch              | add TypeAbstractions as a language extension | yes           |
+| singletons-base-3.1.1.patch         | add TypeAbstractions as a language extension | yes           |
+| vector-space-0.16.patch             | * is type (4)                                | yes           |
+
+`th-compat-0.1.4.patch` was miscounted so I left that out.
+Simplified subsumption appears a lot but 3 are for Cabal,
+so it's only 2 real occurrences.
+We expect that to appear a lot however,
+because it was one of *the* motivating changes for a [stability working group](https://blog.haskell.org/stability-working-group/).
+
+## Simplified subsumption
+For the blissfully ignorant reader simplified subsumption causes you
+to do this under certain existential conditions:
+```haskell
```

Comment:
If you want to highlight the changes instead of the syntax, you should use `diff` instead of `haskell`. This will help the readers understand what has actually changed.